manual/01-intro/Immutability.md

```diff
@@ -1,4 +1,168 @@
 Immutability
 ============
 
-TODO
+Often in Javascript we pair field accessors and field mutators for our
+data structures.  We might see
+
+```js
+person.setName('Fred');
+person.getName(); //=> "Fred"
+```
+
+This is done in many different ways.  Of course fields themselves are
+most commonly both readable and writable:
+
+```js
+person.name = 'Fred';
+person.name; //=> "Fred"
+```
+
+And there are more convoluted means involving `Object.defineProperty`.
+In each of these techniques, there is a straightforward pairing.  Those
+things you can _read_ from an object, you can also _write_ to it.
+
+In functional programming, such mutable state is considered extremely
+dangerous, making programs hard to reason about, difficult to debug,
+and almost impossible to grow correctly into large, consistent systems.
+
+Ramda helps avoid this by refusing to ever mutate input data.  We have
+equivalents of the accessors above:
+
+```js
+var getName = R.prop('name');
+getName(person); //=> "Fred"
+// or 
+R.prop('name', person); //=> "Fred"
+```
+
+But the closest equivalents we have to mutators are functions like 
+`assoc`.  And they're not all that close.  `assoc` creates a copy of
+your object with a new value associated with a field name:
```

Comment:
It would be nice for the first reference to each function in each section to link to the function's documentation.